metadata_updater/metadata_updater.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/python3
+################################################################################
+#
+# Copyright 2018 Crown copyright (c)
+# Land Information New Zealand and the New Zealand Government.
+# All rights reserved
+#
+# This program is released under the terms of the new BSD license. See the
+# LICENSE file for more information.
+#
+################################################################################
+
+import koordinates
+import os
+import sys
+import fileinput
+import yaml
+import re
+import requests
+import logging
+
+try:
+    import log
+except:
+    from . import log
+
+logger = logging.getLogger(__name__)
+ERRORS = 0
+
+class ConfigReader():
+    """
+    Create a config object that can
+    be passed to the scripts methods
+    """
+
+    def __init__(self):
+        # READ CONFIG
+        with open(os.getcwd()  + '/config.yaml', 'r') as f:
+            config = yaml.load(f)
+
+        if "Connection" in config:
+            self.api_key = config["Connection"]["Api_key"]
```

Comment:
Would be safer to support API key as an environment variable, so people don't have to save it in plain text file.

Reply:
API Keys now supported as environmental variable. This must be named LDS_APIKEY. Documentation has been updated

Though. users can choose to add their key to the config instead. The code will check if there is a LDS_APIKEY envi var else try the config. I found the config very useful for testing API keys quickly with different scope to ensure the scope required was documented correctly. There is little need for this Api_key config.yml property now. I am open to removing it and forcing users to use the envi var. Thoughts? or shall the user just make a decision about risk

Reply:
I think to keep it simple I would remove it from the config file. If the user really wants to store their API key in a file they could insert it into the command line option with stdin direct.